src/diagnostic-channel-publishers/src/mongodb.pub.ts

```diff
@@ -96,16 +96,98 @@ const mongodb3PatchFunction: PatchFunction = function(originalMongo) {
     return originalMongo;
 };
 
+// In mongodb 3.3.0, mongodb-core was merged into mongodb, so the same patching
+// can be used here. this.s.pool was changed to this.s.coreTopology.s.pool
+const mongodbcorePatchFunction = function(originalMongo) {

```

Comment:
This patching is duplicate from what is in mongodb-core's patching, so I'm wondering if this is fine or if it should instead just call something in core's patching. Or have an additional patch setup in core which patches `mongodb` for 3.3 --> 4.0 (e.g. patching part of it in this file, and the old "core" logic in core)

Reply:
I believe having duplicated code here is fine because is easier to understand, moving forward this will make more sense instead of using old stuff code(core)